pinot-common/src/main/java/org/apache/pinot/common/datablock/BaseDataBlock.java

```diff
@@ -338,6 +339,14 @@ public String[] getStringArray(int rowId, int colId) {
     return strings;
   }
 
+  @Override
+  public Map<String, Object> getMap(int rowId, int colId) {
+    int size = positionOffsetInVariableBufferAndGetLength(rowId, colId);
+    ByteBuffer buffer = _variableSizeData.slice();
+    buffer.limit(size);
+    return MapUtils.deserializeMap(buffer);
```

Comment:
It could be expensive for general map conversion and it's ok to be slow for select the whole map column. 
The major purpose of introducing MAP type is to support more indexing for filtering and aggregation.